### ISSUE TITLE
argument destructuring and rest params triggers no-redeclare rule twice.

### ISSUE THREAD

**Issue by BerkeleyTrue** [eslint/eslint#4513]
eslint v1.10.1 (seen in 1.9.0 as well)

Code
```js
challenges.map(({ tests, ...rest }) => {
  const solved = tests.every(test => !test.err);
  return { ...rest, tests, solved };
})
```

Expected no lint warnings.
Actually got lint warnings:
```bash
50:27  warning  "rest" is already defined  no-redeclare
50:27  warning  "rest" is already defined  no-redeclare
```

**Comment by eslintbot** [comment]
@BerkeleyTrue Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Are you using babel-eslint?

**Comment by beenokle** [comment]
We are faced with the same issue
eslint 1.10.1
babel-eslint 4.1.5

**Comment by nzakas** [comment]
Duplicate of #4481. Please reinstall ESLint.
